src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1067,24 +1068,30 @@ private function registerSerializerConfiguration(array $config, ContainerBuilder
         }
 
         $chainLoader->replaceArgument(0, $serializerLoaders);
+        $container->getDefinition('serializer.mapping.cache_warmer')->replaceArgument(0, $serializerLoaders);
+
+        $cache = null;
 
         if (isset($config['cache']) && $config['cache']) {
             @trigger_error('The "framework.serializer.cache" option is deprecated since Symfony 3.1 and will be removed in 4.0. Configure the "cache.serializer" service under "framework.cache.pools" instead.', E_USER_DEPRECATED);
 
+            $cache = new Definition(DoctrineProvider::class, array(new Reference($config['cache'])));
```

Comment:
Isn't this a BC break?

Before: the old factory is used with the Doctrine cache as second parameter. PSR-6 / cache component isn't used at all.
After: the factory decorator using PSR-6 and the cache component are used (they still fallback to Docrtrine).

I would avoid this to prevent weird behaviors.